plugins/woocommerce/src/Internal/Traits/RestApiCache.php

```
@@ -0,0 +1,400 @@
+		$request_hash    = md5( implode( '-', $cache_key_parts ) );
+		return "wc_rest_api_cache_{$entity_type}-{$request_hash}";
+	}
+
+	/**
+	 * Get a cached response, but only if it's valid (otherwise the cached response will be invalidated).
+	 *
+	 * @param WP_REST_Request $request       The request object.
+	 * @param array           $cached_config Built caching configuration from build_cache_config().
+	 * @return WP_REST_Response|null Cached response, or null if not available or has been invalidated.
+	 */
+	private function get_cached_response( WP_REST_Request $request, array $cached_config ): ?WP_REST_Response {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)